ml/synthesis/src/scripts/evaluation.py

```diff
@@ -0,0 +1,353 @@
+import asyncio
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import List, Optional, Any
+from cached_property import cached_property
+from cachetools import cached
+from glob import glob
+from IPython.display import display
+
+import argparse
+import subprocess
+import time
+import pandas as pd
+import tensorflow as tf
+import datetime
+
+from aiohttp import ClientSession
+from dataclasses_json import dataclass_json, LetterCase
+
+
+NITTA_PORT = 8080
+NITTA_BASEURL = f"http://localhost:{NITTA_PORT}"
+NITTA_ROOT_DIR = r'../../../..'
+print(NITTA_ROOT_DIR)
+# may needed to be updated
+NITTA_EXE_PATH = "stack exec nitta -- " # NITTA_ROOT_DIR + r"\.stack-work\dist\29cc6475\build\NITTA\nitta.exe"
+METRICS_WEIGHTS = pd.Series(dict(duration=-1, depth=-0.1))
+LAMBDA = 0.6
+WAIT_NITTA_DELAY = 0.5
+pd.set_option('display.max_colwidth', 120)
+
+parser = argparse.ArgumentParser(description="NITTA Tree Evaluator")
+parser.add_argument("-e", "--example", required=True, help="Path to the example file (e.g., examples/teacup.lua)")
+parser.add_argument("-n", "--model_name", required=True, help="Path to Model name (e.g., model_N)")
+parser.add_argument("-r", "--models_root", required=True, help="Path to modelst root (e.g., models)")
+parser.add_argument("-d", "--data_root", required=True, help="Path to data root (e.g., data)")
+args = parser.parse_args()
+
+def cached_node_method(wrapped):
+    return cached({}, key=lambda self, *args: hash(self.sid))(wrapped)
+    
+nitta_dataclass_params = dict(letter_case=LetterCase.CAMEL)
+
+
+def debounce(s):
+    """Decorator ensures function that can only be called once every `s` seconds.
+    """
+    def decorate(f):
+        t = None
+        n = 0
+
+        def wrapped(*args, **kwargs):
+            nonlocal t, n
+            t_ = time.time()
+            if t is None or t_ - t >= s:
+                if n > 0: print(f"-- skipped {n} calls")
+                result = f(*args, **kwargs)
+                t = time.time()
+                n = 0
+                return result
+            else:
+                n += 1
+        return wrapped
+    return decorate
+
+@debounce(1)
+def log_debug(*args):
+    print("--", datetime.datetime.now().strftime("%T"), *args)
+
+
+@dataclass_json(**nitta_dataclass_params)
+@dataclass
+class NittaNodeDecision:
+    tag: str
+    
+@dataclass_json(**nitta_dataclass_params)
+@dataclass
+class NittaNode: 
```

Comment:
already defined in `components`, so we better have it imported it from there and reused :)